src/content/get-started/ch2-add-commands.md

```diff
@@ -0,0 +1,426 @@
+    void main(List<String> arguments) {
+      if (arguments.isNotEmpty && arguments.first == 'version') {
+        print('Dartpedia CLI version $version');
+      } else if (arguments.isNotEmpty && arguments.first == 'help') {
+        printUsage();
+      } else {
+        printUsage(); // Change this from 'Hello, Dart!'
+      }
+    }
+    ```
+
+5.  **Understand the `if/else` structure and variables:** You've now built an
+    `if`, `else if`, and `else` structure for conditional execution.
+
+    * `arguments.isNotEmpty` checks if any command-line arguments were
+        provided.
+    * `arguments.first` accesses the very first argument, which we're using as
+        our command.
+    * Notice the `version` variable is declared as a `const`. This means its
+        value is known at compile time and you cannot change it during runtime.
+    * `arguments`, on the other hand, is a regular (non-constant) variable
+        because its content can change based on user input.
+
+    Test the `help` command: Run your application with the help argument:
+
+    ```bash
+    dart bin/cli.dart help
+    ```
+
+    You should see the usage information printed.
+
+    Also, try running it without any arguments:
+
+    ```bash
+    dart bin/cli.dart
+    ```
+
+    It should now also display the usage information.
+
+    :::note
+    Any command-line argument that is not `version` or `help` will currently
+    also print the usage information. This is expected behavior for now.
+    :::
+
+### Task 2: Implement the search command
+
+Next, implement a basic `search` command that takes an article title as
+input. As you build this functionality, you'll work with `List` manipulation,
+null checks, and string interpolation.
+
+1.  **Integrate the `search` command into `main`:** First, modify your `main`
+    function in `cli/bin/cli.dart` to include an `else if` branch that handles
+    the `search` command. For now, we'll just print a placeholder message.
+
+    ```dart
+    void main(List<String> arguments) {
+      if (arguments.isNotEmpty && arguments.first == 'version') {
+        print('Dartpedia CLI version $version');
+      } else if (arguments.isNotEmpty && arguments.first == 'help') {
+        printUsage();
+      } else if (arguments.isNotEmpty && arguments.first == 'search') {
+        // Add this new block:
+        print('Search command recognized!');
+      } else {
+        printUsage();
+      }
+    }
+    ```
+
+2.  **Test the new command:** Run your application with the `search`
+    command:
+
+    ```bash
+    dart bin/cli.dart search
+    ```
+
+    You should see:
+
+    ```bash
+    Search command recognized!
+    ```
+
+3.  **Define the `runApp` function:** Your `search` command will eventually run
+    the core logic of our application. Next, create a new function called
+    `runApp` that will house this logic. Initially, `runApp` will just take a
+    list of arguments and print them. Place this new function below `main`.
```

Comment:
```suggestion
3.  **Define the `runApp` function:** The `search` command will eventually run
    the core logic of your application by calling a function called `runApp`. For now, 
    it will print the arguments passed into search. Place this new function below `main`.
```